Modules/Common/UTXOTraits.php

```diff
@@ -163,9 +165,20 @@ private static function bc_hexdec($num)
 }
 
 // Carefully converts floats from the node to numerics
-function satoshi($value)
+function satoshi(string $value, CoreModule|false $module = false): string
 {
-    if ($value === '0.00000000') return '0';
-    if (strlen(explode('.', $value)[1]) !== 8) throw new \ModuleError("satoshi(value: ({$value})): incorrect value");
-    return ltrim(str_replace('.', '', $value), '0');
+    if ($module === false || !in_array(UTXOSpecialFeatures::Not8Decimals, $module->extra_features))
+    {
+        if ($value === '0.00000000') return '0';
+        if (strlen(explode('.', $value)[1]) !== 8) throw new ModuleError("satoshi(value: ({$value})): incorrect value");
+        return ltrim(str_replace('.', '', $value), '0');
+    }
+    else
+    {
+        $decimals = $module->currency_details['decimals'];
+        if ($value === '0.' . str_repeat('0', $decimals)) return '0';
+        $value_parts = explode('.', $value);
+        if (strlen($value_parts[1]) !== $decimals) throw new ModuleError("satoshi(value: ({$value})): incorrect number of decimals");
```

Comment:
For the history: error throwing kept to catch the case and be aware when this happens

> [The Peercoin’s system is the equivalent of allowing an unlimited number of decimals](https://talk.peercoin.net/t/why-was-ppc-designed-to-be-inflationary/316/8)